test/e2e/e2e_suite_test.go

```diff
@@ -101,3 +105,29 @@ func initScheme() *runtime.Scheme {
 	framework.TryAddDefaultSchemes(scheme)
 	return scheme
 }
+
+func dumpLogs() {
+	pods := &corev1.PodList{}
+	Eventually(func() (bool, error) {
+		if err := kubeClient.List(context.TODO(), pods, client.InNamespace(corev1.NamespaceDefault)); err != nil {
+			return false, err
+		}
+
+		return true, nil
+	}, framework.ListTimeout, framework.ListPolling).Should(BeTrue())
+
+	var micPods, nmiPods []corev1.Pod
+	for _, pod := range pods.Items {
```

Comment:
In general, I think more scalable option would be just do a filtered list with MIC and NMI pod labels